src/lightning-address.ts

```diff
@@ -64,6 +67,41 @@ export default class LightningAddress {
     return `https://${this.domain}/.well-known/keysend/${this.username}`;
   }
 
+  async requestInvoice(
+    amount: number,
+    comment: string
+  ): Promise<Invoice> {
+    await this.fetch();
+    const { callback, commentAllowed, min, max } = parseLnUrlPayResponse(this.lnurlpData);
+    if (!isValidAmount({ amount: amount / 1000, min, max }))
+      throw new Error('Invalid amount')
+  
+    if (!isUrl(callback)) throw new Error('Callback must be a valid url')
+
+    let callbackUrl = new URL(callback)
+
+    const invoiceParams: {amount: string, comment?: string} = { amount: amount.toString() };
+
+    if (comment && commentAllowed > 0 && comment.length > commentAllowed)
+    throw new Error(
+      `The comment length must be ${commentAllowed} characters or fewer`
+    )
+    if (comment) invoiceParams.comment = comment
+
+    callbackUrl.search = new URLSearchParams(invoiceParams).toString()
+  
+    const data = await fetch(callbackUrl)
+    const json = await data.json()
+    const paymentRequest = json && json.pr && json.pr.toString()
+    const verifyURL = json && (json.verify || '').toString()
```

Comment:
does this need to be a sting or can it be undefined/null?

Reply:
it can be undefined or null

Reply:
yes, then we write an empty string in `verifyURL` ? is that a good practice?